rotkehlchen/data_import/manager.py

```diff
@@ -0,0 +1,52 @@
+from pathlib import Path
+from typing import Any, Tuple
+
+from rotkehlchen.data_import.importers.binance import BinanceImporter
+from rotkehlchen.data_import.importers.bisq_trades import BisqTradesImporter
+from rotkehlchen.data_import.importers.blockfi_trades import BlockfiTradesImporter
+from rotkehlchen.data_import.importers.blockfi_transactions import BlockfiTransactionsImporter
+from rotkehlchen.data_import.importers.cointracking import CointrackingImporter
+from rotkehlchen.data_import.importers.cryptocom import CryptocomImporter
+from rotkehlchen.data_import.importers.nexo import NexoImporter
+from rotkehlchen.data_import.importers.shapeshift_trades import ShapeshiftTradesImporter
+from rotkehlchen.data_import.importers.uphold_transactions import UpholdTransactionsImporter
+from rotkehlchen.data_import.utils import BaseExchangeImporter
+from rotkehlchen.db.dbhandler import DBHandler
+from rotkehlchen.types import DataImportSource
+
+
+class CSVDataImporter:
+    """This class is responsible for importation of csv files."""
+    def __init__(self, db: DBHandler):
+        self.db = db
+
+    def import_csv(
+            self,
+            source: DataImportSource,
+            filepath: Path,
+            **kwargs: Any,
+    ) -> Tuple[bool, str]:
+        """Imports csv data from `filepath`.`source` determines the format of the file.
+        Returns (True, '') if imported successfully and (False, message) otherwise."""
+        importer: BaseExchangeImporter
+        if source == DataImportSource.COINTRACKING:
+            importer = CointrackingImporter(db=self.db)
+        elif source == DataImportSource.CRYPTOCOM:
+            importer = CryptocomImporter(db=self.db)
+        elif source == DataImportSource.BLOCKFI_TRANSACTIONS:
+            importer = BlockfiTransactionsImporter(db=self.db)
+        elif source == DataImportSource.BLOCKFI_TRADES:
+            importer = BlockfiTradesImporter(db=self.db)
+        elif source == DataImportSource.NEXO:
+            importer = NexoImporter(db=self.db)
+        elif source == DataImportSource.SHAPESHIFT_TRADES:
+            importer = ShapeshiftTradesImporter(db=self.db)
+        elif source == DataImportSource.UPHOLD_TRANSACTIONS:
+            importer = UpholdTransactionsImporter(db=self.db)
+        elif source == DataImportSource.BISQ_TRADES:
+            importer = BisqTradesImporter(db=self.db)
+        elif source == DataImportSource.BINANCE:
+            importer = BinanceImporter(db=self.db)
```

Comment:
What about cyclic imports? This should be coded in a way to not have cyclic imports. The current code is still wrong.

Please do what I told you in the previous review. You can move `DataImportSource` anywhere you need.

Reply:
Done